optimum/intel/ipex/inference.py

```diff
@@ -188,16 +101,23 @@ def __enter__(self):
                             with torch.cpu.amp.autocast(enabled=(self._dtype == torch.bfloat16)), torch.no_grad():
                                 if self._model.tokenizer is not None and self._jit:
                                     try:
-                                        jit_inputs = prepare_jit_inputs(self._model.model, self._model.task)
-                                        model = torch.jit.trace(model, jit_inputs, strict=False)
-                                        model = torch.jit.freeze(model)
-                                        model(*jit_inputs)
-                                        model(*jit_inputs)
+                                        jit_model = jit_trace(
+                                            model=model,
+                                            task=self._model.task,
+                                            use_cache=self._original.config.use_cache,
+                                        )
+                                        if self._model.task == "text-generation":
+                                            model = TSModelForCausalLM(
+                                                model=jit_model,
+                                                config=self._original.config,
+                                                use_cache=self._original.config.use_cache,
+                                            )
                                     except Exception as e:
+                                        self._jit = False
                                         logger.warning(f"failed to use PyTorch jit mode due to: {e}.")
                                 # Patching model with the new one
-                                if self._model.task == "text-generation":
-                                    self._model.model = _ModelGenerationWrapper(model, self._original)
+                                if self._model.task == "text-generation" and self._jit:
+                                    self._model.model = model

```

Comment:
Shouldn't it be for all cases :
`self._model.model = _ModelFallbackWrapper(model, self._original)`

Reply:
Hi @echarlaix . If I use _ModelFallbackWrapper, when I execute `model.generate`, it will go to the func  `__getattr__(self, item)`, and will return `getattr(self._default, item)` so I actually execute `self._default.generate`. I didn't use _ModelFallbackWrapper because I cannot use the generation of my optimized model.

Reply:
Why not subclass from `_ModelFallbackWrapper` to enable it then ?

Reply:
@echarlaix , I think we should enhance TSModelForCausalLM for all text-generation tasks, then for text-generation task, we don't need to use _ModelFallbackWrapper, is it OK?